KIPs/kip-82.md

```diff
@@ -0,0 +1,376 @@
+If the amount of gas fee that is provided as a basic reward does not exceed a certain level, all of them will be burnt. The threshold amount is 20% of minting provided as a basic reward. If the gas fee exceeds this amount, 20% will be incinerated and the additional amount will be distributed to GC.
+
+To reduce the KLAY supply amount due to inflation, the gas fee will be burnt.
+
+### Klaytn node update
+
+The Klaytn node will be updated according to the new reward policy. The CN portion is further split into basic reward and staked reward. The staked reward is then distributed among CNs proportional to their staking shares. Their shares are calculated proportionally to their stake amounts except for the minimum staking amount.
+
+Two new governance parameters fine control the new reward distribution algorithm. The default values below will be used in the Cypress mainnet. The parameters can be updated by the existing governance mechanism.
+
+| Name                | Type   | Meaning                                              | Default Value |
+|---------------------|--------|------------------------------------------------------|---------------|
+| `reward.usekip82`   | bool   | Whether the new algorithm is used or not             | false         |
+| `reward.kip82ratio` | string | The relative ratio between basic and staking rewards | "20/80"       |
+
+During integer arithmetics, minuscule remaining amounts may be emitted as by-products. Such remaining amounts will be added to the proposer or KGF portion. Below pseudocode illustrates the new reward distribution algorithm. Note: `//` is round down integer division.
+
+```python
+from collections import defaultdict
+
+MAGMA_BLOCK_NUMBER = 99841497
+FORK_BLOCK_NUMBER = 120000000 # TBD
+
+# Block header. Only relevant fields are shown here.
+class Header:
+    Number: int = 0   # Block number
+    GasUsed: int = 0  # Total gas spent in the block
+    BaseFee: int = 0  # Base fee per gas at the block
+    RewardBase: string = "" # Reward recipient address of the block proposer
+
+# Network configurations related to reward distribution.
+# Corresponds to Klaytn's reward.rewardConfig struct.
+class RewardConfig:
+    UnitPrice: int = 25000000000              # "governance.unitprice" parameter (in peb)
+    MintingAmount: int = 9600000000000000000  # "reward.mintingamount" parameter (in peb)
+    MinimumStake: int = 5000000               # "reward.minimumstake" parameter (in KLAY)
+    DeferredTxFee: bool = true    # "reward.deferredtxfee" parameter
+    UseKIP82: bool = true         # "reward.usekip82" parameter (new)
+
+    # "reward.ratio" parameter (e.g. "34/54/12")
+    CnRatio: int = 34
+    KgfRatio: int = 54
+    KirRatio: int = 12
+    TotalRatio: int = CnRatio + KgfRatio + KirRatio
+
+    # "reward.kip82ratio" parameter (e.g. "20/80") (new)
+    CnMintedBasicRatio: int = 20
```

Comment:
@blukat29 we need to define terminology. In the figure it is denoted as GC block proposer reward, but in the code it is Basic Ratio. Which one do you prefer?

Reply:
```suggestion
    ProposingRewardRatio: int = 20
```